doc/fgpu.design.rst

```diff
@@ -259,14 +339,121 @@ written as :math:`pm + s`. Let :math:`z^r` denote the array :math:`z_r, z_{n+r},
-dimension with corresponding stride. We solve this by modifying the PFB kernel
-to reorder its output such that each :math:`z^r` is output contiguously. This
-can be done by shuffling some bits in the output index (because we assume
-powers of two everywhere).
+Now let's walk through each part. To recap the indexing used in the Cooley-Tukey algorithm: let a
+time-domain index :math:`i` be written as :math:`qn + r` and a frequency-domain index :math:`k` be
+written as :math:`pm + s`. Let :math:`z^r` denote the array :math:`z_r, z_{n+r}, \dots, z_{(m-1)n+r}`,
+and denote its Fourier transform by :math:`Z^r`. It is worthwhile to point out that the superscript
+:math:`r` *does not* denote raise to the power but rather is a means to indicate an :math:`r^{th}` array.
+In practice this :math:`r^{th}` array is a subset (part) of the larger :math:`z` array of input data.
+
+Let's unpack this a bit further --- what is actually happening
+is that the initial array :math:`z` is divided into :math:`n=4` separate arrays each of
+:math:`m=32768/4 = 8192` elements (hence the :math:`N = mn` above)(This is true for a desired overall
+transform size of 32768 channels). The actual samples that land up in each array are defined by the
+indices :math:`i` and :math:`k`. Lets start with :math:`i`.
+
+It was stated that :math:`i = qn + r`. The parameter :math:`r` takes on the range :math:`0` to :math:`n-1`
+(so :math:`r=0` to :math:`r=3` as :math:`n = 4`) and :math:`q` takes on the
+range :math:`0` to :math:`m-1` (i.e. :math:`q=0` to :math:`q=8191`).
+So we are dividing up array :math:`z` into :math:`n` smaller arrays denoted by  :math:`r` (i.e. :math:`z^{r}`)
+each of length :math:`m=8192`. So what does this look like practically?
+
+The first array when :math:`r=0` (i.e. :math:`z^{0}`)
+
+========================   ========
+   Inputs                  Index(i)
+------------------------   --------
+  qn + r                      i
+========================   ========
+:math:`0 \cdot 4` + 0         0
+:math:`1 \cdot 4` + 0         4
+:math:`2 \cdot 4` + 0         8
+...               ...
+...               ...
+:math:`8191 \cdot 4` + 0    32764
+========================   ========
+
+... (intentionally skipping 2nd and 3rd array to conserve space)
+
+The fourth array when :math:`r=3` (i.e. :math:`z^{3}`)
+
+========================  ========
+   Inputs                 Index(i)
+------------------------  --------
+  qn + r                     i
+========================  ========
+:math:`0 \cdot 4 + 3`        3
+:math:`1 \cdot 4 + 3`        7
+:math:`2 \cdot 4 + 3`        11
+...               ...
+...               ...
+:math:`8191 \cdot 4 + 3`   32767
+========================  ========
+
+What this shows is that each sub-array consists of samples from the initial array :math:`z` indexed
+by :math:`i=qn+r` where each sample is every :math:`4^{th}` and offset by :math:`r`. Pictorially
+this looks like,
+
+.. image:: images/z_array.png
```

Comment:
Where is the source for this image?

Reply:
The image still references x rather than z; and the caption in the middle makes it sound like z^r are the outputs of a Fourier transform, rather than the inputs.

Reply:
Previous feedback suggested the image be modified to x_0 to x_65535. I'll commit the source (.ppt).

Previous feedback:
The top line refers to z, but it should refer to x, and the values should be labelled x_0 to x_65535, not r_0 to r_65536.
It shows the z^r arrays each containing either only real or only imaginary components. These are complex arrays. At this stage we're not worrying about the real-to-complex transform, and I think the diagram would be less confusing if it only dealt with z and showed complex numbers without trying to split them into real and imaginary.

Reply:
Image updated as requested.